tools/ldb_cmd.cc

```diff
@@ -1812,8 +1816,19 @@ void InternalDumpCommand::DoCommand() {
 
     if (!count_only_ && !count_delim_) {
       std::string key = ikey.DebugString(is_key_hex_);
-      std::string value = Slice(key_version.value).ToString(is_value_hex_);
-      std::cout << key << " => " << value << "\n";
+      Slice value = Slice(key_version.value);
+      if (!decode_blob_index_ || value_type != kTypeBlobIndex) {
+        std::cout << key << " => " << value.ToString(is_value_hex_) << "\n";
+      } else {
+        BlobIndex blob_index;
+
+        const Status s = blob_index.DecodeFrom(value);
+        if (!s.ok()) {
+          fprintf(stderr, "%s => error decoding blob index =>\n", key.c_str());
+        } else {
+          std::cout << key << " => " << blob_index.DebugString(is_key_hex_);
```

Comment:
I think we would want to pass `is_value_hex_` to `DebugString` here since `DebugString` uses it to control how the (inlined) value is printed.

Reply:
Good catch, it should use the bool for the value one instead.